### PR TITLE
add current pose to the local planner critics' prepare() method

### PR DESCRIPTION
**Pull request by ChristofDubs**
preparation for [DWA planner: report start blocked](https://www.wrike.com/open.htm?id=1020463025) and [DWA planner: report path blocked](https://www.wrike.com/open.htm?id=1020463048)

### CODE REVIEW

File: base_local_planner/include/base_local_planner/trajectory_cost_function.h [hunk 1]
```diff
@@ -58,7 +58,7 @@ class TrajectoryCostFunction {
    * General updating of context values if required.
    * Subclasses may overwrite. Return a non-zero error code in case there is any error.
    */
-  virtual ExePathOutcome prepare() = 0;
+  virtual ExePathOutcome prepare(const geometry_msgs::PoseStamped& current_pose) = 0;
```
Comment (renan028) [inline base_local_planner/include/base_local_planner/trajectory_cost_function.h:62]:
```suggestion
  virtual ExePathOutcome prepare([[maybe_unused]] const geometry_msgs::PoseStamped& current_pose) = 0;
```

File: base_local_planner/include/base_local_planner/trajectory_search.h [hunk 2]
```diff
@@ -57,7 +57,7 @@ class TrajectorySearch {
    * @param traj The container to write the result to
    * @param all_explored pass NULL or a container to collect all trajectories for debugging (has a penalty)
```
Comment (renan028) [inline base_local_planner/include/base_local_planner/trajectory_search.h:59]: F: add new param description

**Review by renan028 — APPROVED** [review]

**Review by corot — APPROVED** [review]
doc for the new param missing

File: base_local_planner/include/base_local_planner/trajectory_cost_function.h (hunk 1 above)
Comment (ChristofDubs) [reply base_local_planner/include/base_local_planner/trajectory_cost_function.h:62]: Hmm I thought we keep this all old style c++ code to minimize changes made compared to upstream; `[[maybe_unused]]` is c++ 17 and we aren't setting `set(CMAKE_CXX_STANDARD 17)` anywhere either?

Comment (renan028) [reply base_local_planner/include/base_local_planner/trajectory_cost_function.h:62]: ah ok, got it

**Pull request merged**